### PR TITLE
Update configuration-reference.md

### LINKED ISSUE #3006: Reusing Config is not linked for commands documentation
# Missing Information
The Reusing Config and some subsection within it is not linked from the commands section.  This is very frustrating when trying to find the syntax for the parameters section.

The reference documentation is similarly not linked to from the orb intro which again is frustrating.  The whole orb documentation should probably be reviewed and concepts linked into the reference section.

The "steps" parameter to the "commands" just says "A sequence of steps run inside the calling job of the command." without linking to the reference spec for steps.

Etc..

### PR DESCRIPTION
**Pull request by thomasr8**
Added links to content as needed based on feedback. This fixes #3006

# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/configuration-reference.md [hunk 1]
```diff
@@ -1126,7 +1093,7 @@ A job may have a `type` of `approval` indicating it must be manually approved be
 ```
 **Note:** The `hold` job name must not exist in the main configuration.
 
-###### **`filters`**
+##### **`filters`**
```
Comment (michelle-luna) [inline jekyll/_cci2/configuration-reference.md]: this TOC level should stay six, not go to five (in the TOC we mimic that tabbing that should be used int he YAML)

File: jekyll/_cci2/configuration-reference.md [hunk 2]
```diff
@@ -1173,10 +1136,9 @@ only | N | String, or List of Strings | Either a single tag specifier, or a list
 ignore | N | String, or List of Strings | Either a single tag specifier, or a list of tag specifiers
 {: class="table table-striped"}
 
-For more information,
-see the [Executing Workflows For a Git Tag]({{ site.baseurl }}/2.0/workflows/#executing-workflows-for-a-git-tag) section of the Workflows document.
+For more information, see the [Executing Workflows For a Git Tag]({{ site.baseurl }}/2.0/workflows/#executing-workflows-for-a-git-tag) section of the Workflows document.
 
-###### *Example*
+##### *Example*
```
Comment (michelle-luna) [inline jekyll/_cci2/configuration-reference.md]: This should also stay at level 5

File: jekyll/_cci2/configuration-reference.md (hunk 1 above)
Comment (thomasr8) [reply jekyll/_cci2/configuration-reference.md]: Fixed

File: jekyll/_cci2/configuration-reference.md (hunk 2 above)
Comment (thomasr8) [reply jekyll/_cci2/configuration-reference.md]: Fixed

**Pull request merged**